gated-nft/zksync/test/main.test.ts

```diff
@@ -1,37 +1,239 @@
 import { expect } from "chai";
-import { Wallet, Provider, Contract } from "zksync-web3";
-import * as hre from "hardhat";
-import { Deployer } from "@matterlabs/hardhat-zksync-deploy";
+import { ERC721, ERC721GatedPaymaster, Greeter } from "./utils/utils";
+import { localConfig } from "../../../tests/testConfig";
 
-// load env file
-import dotenv from "dotenv";
-dotenv.config();
+describe("Gated NFT", function () {
+  describe("ERC721", function () {
+    let contract: any;
+    let result: any;
+    const utils = new ERC721();
 
-const PRIVATE_KEY = process.env.WALLET_PRIVATE_KEY || "";
+    beforeEach(async function () {
+      contract = await utils.deployERC721Contract();
+    });
 
-if (!PRIVATE_KEY)
-  throw "⛔️ Private key not detected! Add it to the .env file!";
+    it("Should be deployed the ERC721 contract and return the NFT address", async function () {
+      result = await utils.deployERC721Script();
+      expect(result[0]).to.contain("0x");
```

Comment:
I'd also assert for the length of the address string to match the expected length of an address string